packages/plugins/i18n/server/src/bootstrap.ts

```diff
@@ -14,6 +14,30 @@ const registerModelsHooks = () => {
       await getService('permissions').actions.syncSuperAdminPermissionsWithLocales();
     },
   });
+
+  const i18nModelUIDs = Object.values(strapi.contentTypes)
+    .filter((contentType) => getService('content-types').isLocalizedContentType(contentType))
+    .map((contentType) => contentType.uid);
+
+  strapi.documents.use(async (context, next) => {
+    // @ts-expect-error ContentType is not typed correctly on the context
+    const schema = context.contentType;
+
+    if (!['create', 'update'].includes(context.action)) {
+      return next(context);
+    }
+
+    if (!i18nModelUIDs.includes(schema.uid)) {
+      return next(context);
+    }
+
+    // Collect the result of the document service action and sync non localized
+    // attributes based on the response
+    const result = (await next(context)) as any;
```

Comment:
result might not contain all the non localizedAttributes (you can pass a populate that would only retrun some nested data and not all of it) we might need to fetch the data with a deepPopulate to be able to copy correctly (only  populate the fields you need to copy of course)

Reply:
can we have nested non localized attributes? I just checked and compos can not !

Reply:
Maybe we need to be sure those attributes are in the top level "fields"

Reply:
nope it stops at the compo level, the entire compo is localized or not :)

Reply:
then do we need to deep populate here?

Reply:
you need to copy the deep data still yes

Reply:
which deep data would need to be syncronized?

Reply:
discussed with alex, did not know the component itself could be localized, so I yep, populate is required!